From 5beb1e4847c0883ed58d8eb0b06f50eb55d4aa89 Thu Sep 03 17:15:24 2015
From: Yuzhi Zheng <yuzhi@fb.com>
Date: Thu, 03 Sep 2015 17:15:24 -0700
Subject: [PATCH] Renaming RelayRecordState to RelayRecordState

Summary: I want to reuse the name `RelayRecordState` for whether a node is part of a optimistic mutation or if an mutation on the node has resulted in an error. These are currently stored under `__status__` and `__hasError__`. I plan to consolidate some of the logic in RelayRecordState. Renaming the current RecordState to State since node has to be in one and only one of these states (I'm thinking state machine). Want to resuse `RelayRecordState` since a record is capable of multiple statuses.

Reviewed By: @josephsavona

Differential Revision: D2412702
---

diff --git a/src/store/RelayRecordState.js b/src/store/RelayRecordState.js
new file mode 100644
--- /dev/null
+++ b/src/store/RelayRecordState.js
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule RelayRecordState
+ * @flow
+ * @typechecks
+ */
+
+'use strict';
+
+export type RecordState = $Enum<typeof RelayRecordState>;
+
+var RelayRecordState = {
+  /**
+   * Record exists (either fetched from the server or produced by a local,
+   * optimistic update).
+   */
+  EXISTENT: 'EXISTENT',
+
+  /**
+   * Record is known not to exist (either as the result of a mutation, or
+   * because the server returned `null` when queried for the record).
+   */
+  NONEXISTENT: 'NONEXISTENT',
+
+  /**
+   * Record State is unknown because it has not yet been fetched from the
+   * server.
+   */
+  UNKNOWN: 'UNKNOWN',
+};
+
+module.exports = RelayRecordState;
diff --git a/src/store/RelayRecordState.js b/src/store/RelayRecordState.js
deleted file mode 100644
--- a/src/store/RelayRecordState.js
+++ /dev/null
@@ -1,38 +0,0 @@
-/**
- * Copyright 2013-2015, Facebook, Inc.
- * All rights reserved.
- *
- * This source code is licensed under the BSD-style license found in the
- * LICENSE file in the root directory of this source tree. An additional grant
- * of patent rights can be found in the PATENTS file in the same directory.
- *
- * @providesModule RelayRecordState
- * @flow
- * @typechecks
- */
-
-'use strict';
-
-export type RecordState = $Enum<typeof RelayRecordState>;
-
-var RelayRecordState = {
-  /**
-   * Record exists (either fetched from the server or produced by a local,
-   * optimistic update).
-   */
-  EXISTENT: 'EXISTENT',
-
-  /**
-   * Record is known not to exist (either as the result of a mutation, or
-   * because the server returned `null` when queried for the record).
-   */
-  NONEXISTENT: 'NONEXISTENT',
-
-  /**
-   * Record status is unknown because it has not yet been fetched from the
-   * server.
-   */
-  UNKNOWN: 'UNKNOWN',
-};
-
-module.exports = RelayRecordState;
diff --git a/src/store/__mocks__/RelayRecordState.js b/src/store/__mocks__/RelayRecordState.js
new file mode 100644
--- /dev/null
+++ b/src/store/__mocks__/RelayRecordState.js
@@ -0,0 +1,12 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+'use strict';
+
+module.exports = require.requireActual('RelayRecordState');
diff --git a/src/store/__mocks__/RelayRecordState.js b/src/store/__mocks__/RelayRecordState.js
deleted file mode 100644
--- a/src/store/__mocks__/RelayRecordState.js
+++ /dev/null
@@ -1,12 +0,0 @@
-/**
- * Copyright 2013-2015, Facebook, Inc.
- * All rights reserved.
- *
- * This source code is licensed under the BSD-style license found in the
- * LICENSE file in the root directory of this source tree. An additional grant
- * of patent rights can be found in the PATENTS file in the same directory.
- */
-
-'use strict';
-
-module.exports = require.requireActual('RelayRecordState');

--
1.7.9.5
